js/src/main/scala/JSVyxal.scala

```diff
@@ -10,33 +9,35 @@ import scala.scalajs.js.JSConverters.*
 @JSExportTopLevel("Vyxal")
 object JSVyxal:
   @JSExport
-  def execute(code: String, inputs: String, flags: String): String =
+  def execute(
+      code: String,
+      inputs: String,
+      flags: String,
+      printFunc: String
+  ): Unit =
     // todo take flags to set settings
     // todo take functions to print to custom stdout and stderr
-    val output = document.getElementById("output")
-    // cast output object to textarea
-    val textarea = output.asInstanceOf[dom.html.TextArea]
-    textarea.value = ""
-    val settings =
-      Settings(online = true)
+    val settings = flags.foldLeft(Settings(online = true))(_.withFlag(_))
     val globals = Globals(
-      settings = settings
+      settings = settings,
+      printFn = (x: VAny) =>
+        val f = js
```

Comment:
This evals `printFunc` every time you call it. Is that desired?

Reply:
Yes, it is